generics/services/kendra-service.js

```diff
@@ -7,56 +7,112 @@
 
 //dependencies
 
-let urlPrefix = 
-process.env.KENDRA_SERIVCE_HOST + 
-process.env.KENDRA_SERIVCE_BASE_URL +
-process.env.URL_PREFIX; 
+let urlPrefix =
+    process.env.KENDRA_SERIVCE_HOST +
+    process.env.KENDRA_SERIVCE_BASE_URL +
+    process.env.URL_PREFIX;
 
 const request = require('request');
+const fs = require('fs');
 
 /**
-  * Get platform user roles
-  * @function
-  * @name platformUserProfile
-  * @returns {Promise} returns a promise.
-*/
-
-var getPreSignedUrl = function ( token,body,endpoint ) {
-
-    const getPresignedUrl = 
-    urlPrefix + endpoint;
-    
+ * to upload file to uploadFileToCloud
+ * @name uploadFileToCloud
+ * @param {String} filePath filePath of the file to upload
+ * @param {String} uploadPath - location of bucket where to upload
+ * @param {String} bucketName - name of the bucket
+ * @param {String} token - logged in user token
+ * @param {String} endpoint - endpoint of the api 
+ * @returns {json} - upload file details
+ */
+function uploadFileToCloud(filePath, uploadPath, bucketName, token, endpoint) {
     return new Promise(async (resolve, reject) => {
         try {
-
             let options = {
-                "headers":{
-                "content-type": "application/json",
-                "authorization" :  process.env.AUTHORIZATION,
-                "x-authenticated-user-token" : token,
+                "headers": {
+                    'Content-Type': "application/json",
+                    "X-authenticated-user-token": token,
+                    "internal-access-token": process.env.INTERNAL_ACCESS_TOKEN
                 },
-                json : body
+                formData: {
+                    filePath: uploadPath,
+                    bucketName: bucketName,
+                    file: fs.createReadStream(filePath)
+
+                }
             };
-            request.post(getPresignedUrl,options,callback);
-            function callback(err,data){
-                if( err ) {
+
+            let apiUrl =
+                urlPrefix + endpoint;
```

Comment:
@rakeshSgr -We can rename urlPrefix to apiBaseUrl

Reply:
resolved